internal/extension/extension.go

```diff
@@ -57,14 +80,71 @@ func (em *ExtensionManager) checkAgentRunning() {
-			em.isExtensionRunning = true
-		} else {
-			logger.Debug("Will use the API since the Serverless Agent was detected but the hello route was unreachable")
-			em.isExtensionRunning = false
+		logger.Debug("Will use the Serverless Agent")
+		em.isExtensionRunning = true
+	}
+}
+
+func (em *ExtensionManager) SendStartInvocationRequest(ctx context.Context, eventPayload json.RawMessage) context.Context {
+	body := bytes.NewBuffer(eventPayload)
+	req, _ := http.NewRequest(http.MethodPost, em.startInvocationUrl, body)
+
+	if response, err := em.httpClient.Do(req); err == nil && response.StatusCode == 200 {
+		// Propagate dd-trace context from the extension response if found in the response headers
+		traceId := response.Header.Values(string(DdTraceId))
+		if len(traceId) > 0 {
+			ctx = context.WithValue(ctx, DdTraceId, traceId[0])
+		}
+		parentId := response.Header.Values(string(DdParentId))
+		if len(parentId) > 0 {
+			ctx = context.WithValue(ctx, DdParentId, parentId[0])
 		}
+		samplingPriority := response.Header.Values(string(DdSamplingPriority))
+		if len(samplingPriority) > 0 {
+			ctx = context.WithValue(ctx, DdSamplingPriority, samplingPriority[0])
+		}
+	}
+	return ctx
+}
+
+func (em *ExtensionManager) SendEndInvocationRequest(ctx context.Context, functionExecutionSpan ddtrace.Span, err error) {
+	// Handle Lambda response
+	lambdaResponse := ctx.Value(DdLambdaResponse)
+	content, responseErr := json.Marshal(lambdaResponse)
+	if responseErr != nil {
+		content, _ = json.Marshal("{}")
+	}
+	body := bytes.NewBuffer(content)
+
+	// Build the request
+	req, _ := http.NewRequest(http.MethodPost, em.endInvocationUrl, body)
+
+	// Mark the invocation as an error if any
+	if err != nil {
+		req.Header[string(DdInvocationError)] = append(req.Header[string(DdInvocationError)], "true")
+	}
+
+	// Extract the DD trace context and pass them to the extension via request headers
+	traceId, ok := ctx.Value(DdTraceId).(string)
+	if ok {
+		req.Header[string(DdTraceId)] = append(req.Header[string(DdTraceId)], traceId)
+		if parentId, ok := ctx.Value(DdParentId).(string); ok {
+			req.Header[string(DdParentId)] = append(req.Header[string(DdParentId)], parentId)
+		}
+		if spanId, ok := ctx.Value(DdSpanId).(string); ok {
+			req.Header[string(DdSpanId)] = append(req.Header[string(DdSpanId)], spanId)
+		}
+		if samplingPriority, ok := ctx.Value(DdSamplingPriority).(string); ok {
+			req.Header[string(DdSamplingPriority)] = append(req.Header[string(DdSamplingPriority)], samplingPriority)
+		}
+	} else {
+		req.Header[string(DdTraceId)] = append(req.Header[string(DdTraceId)], fmt.Sprint(functionExecutionSpan.Context().TraceID()))
+		req.Header[string(DdSpanId)] = append(req.Header[string(DdSpanId)], fmt.Sprint(functionExecutionSpan.Context().SpanID()))
+	}
+
+	response, err := em.httpClient.Do(req)
+	if response.StatusCode != 200 || err != nil {
+		logger.Debug("Unable to make a request to the extension's end invocation endpoint")
```

Comment:
```suggestion
		logger.Debugf("Unable to make a request to the extension's end invocation endpoint: %v", err)
```

I would also log the error that was returned. Something like this.